Library/Homebrew/formula_installer.rb

```diff
@@ -1074,7 +1074,7 @@ def post_install
       -I #{$LOAD_PATH.join(File::PATH_SEPARATOR)}
       --
       #{HOMEBREW_LIBRARY_PATH}/postinstall.rb
-      #{formula.path}
+      #{formula.prefix/".brew/#{formula.name}.rb"}
```

Comment:
I wonder if this should only use the formula file from the bottle if the tap isn't installed locally (or, rather, the formula isn't found in that tap).

My thinking is that if you make a change to the `post_install` using `brew edit`, it only updates the tap so to really get the changes in a way that's useful you would need to reinstall the entire formula (or edit the formula file which is just a bit more tricky).

Reply:
On the other hand, I guess, maybe the `post_install` details _shouldn't_ change. For example if you have an outdated version and try to run the `post_install` for a new version it might not work as expected, so you _should_ use the bottle block. Is it too complicated to use the tap's `post_install` if the formula and tap exist _and_ the `pkg_version` matches?

So I guess I'm not sure, but throwing it out there

Reply:
The assumption I had was that the formula should be installed to use `postinstall`, and that its contents shouldn't change. However, that is a problem if we want to support editing the formula's `post_install` method and then using `brew postinstall` (I believe we _do_ want to support this, just didn't think about this case in detail).

Currently, `brew postinstall hello` works even if `hello` isn't installed, I'm not sure that's intentional. With this PR in its current state, that would throw a warning that post-install didn't complete successfully, so maybe we also want to make sure `brew postinstall` errors out if the formula isn't installed.

> Is it too complicated to use the tap's `post_install` if the formula and tap exist _and_ the `pkg_version` matches?

Perhaps we could do this. I think it shouldn't be too complicated.

Reply:
> I wonder if this should only use the formula file from the bottle if the tap isn't installed locally (or, rather, the formula isn't found in that tap).

Yes, I agree.



> Is it too complicated to use the tap's `post_install` if the formula and tap exist _and_ the `pkg_version` matches?

This would be nice but don't think it has to block this PR.